ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/__firefox__.js

```diff
@@ -413,22 +413,11 @@ if (!window.__firefox__) {
     return new Promise((resolve, reject) => {
       var oldWebkit = window.webkit;
       delete window['webkit'];
-      
-      // WebKit no longer restores the handler immediately! So we poll for when that happens and resolve the promise accordingly.
-      const timeout = 5000;
-      let startTime = Date.now();
-      
-      // While loop blocks synchronously. SetTimeout or SetInterval can cause a race condition.
-      while(true) {
-        if (window.webkit.messageHandlers && window.webkit.messageHandlers[messageHandlerName]) {
-          let result = window.webkit.messageHandlers[messageHandlerName].postMessage(message);
-          window.webkit = oldWebkit;
-          result.then(resolve).catch(reject);
-          break;
-        } else if (Date.now() - startTime >= timeout) {
-          reject(new TypeError("undefined is not an object (evaluating 'webkit.messageHandlers')"));
-          break;
-        }
+
+      if (window.webkit.messageHandlers && window.webkit.messageHandlers[messageHandlerName]) {
+        let result = window.webkit.messageHandlers[messageHandlerName].postMessage(message);
+        window.webkit = oldWebkit;
+        result.then(resolve).catch(reject);
```

Comment:
Was this intentional revert some of the logic from https://github.com/brave/brave-core/pull/26014? Worth noting I'm seeing the original issue anymore on this branch, with the feature flag enabled or disabled.

Reply:
cc @Brandon-T

Reply:
It's intentional. Discussed in DMs :)